src/sync/impl/sync_metadata.hpp

```diff
@@ -41,6 +41,10 @@ class SyncUserMetadata {
         size_t idx_identity;
         // A locally issued UUID for the user. This is used to generate the on-disk user directory.
         size_t idx_local_uuid;
+        // Another UUID that is guaranteed (as far as UUIDs go) to be unique to this user and device.
+        // Used for partial sync. (Note that `idx_local_uuid` does NOT make this guarantee.)
+        // This property is identical to `idx_local_uuid` if that property does suffice.
```

Comment:
I think it'd help to expand the comment for `idx_local_uuid` to make clear that it may not be a UUID and why (coz we use the user identity when updating from older versions of Realm for compatibility with the existing file system structure?), and then contrast this field with it.